pageserver/src/walingest.rs

```diff
@@ -98,16 +98,24 @@ impl<'a> WalIngest<'a> {
             self.checkpoint_modified = true;
         }
 
-        match decoded.xl_rmid {
+        enum Outcome {
+            Noop,
+            Modified,
```

Comment:
It's not clear to me what has been modified. Maybe `Handled` or `Processed`.

Reply:
> what has been modified

See the `match` and TODO comments later.
The idea is to assert that if `Modified`, the last_record_lsn is the processed WAL record's LSN.